### ISSUE TITLE
OTLPResourceNoServiceName with defined service.name

### ISSUE THREAD

**Issue by johanneswuerbach** [open-telemetry/opentelemetry-cpp#765]
**Describe your environment** 
Using the nginx module configured with a service name and an otlp exporter build from opentelemetry-cpp-contrib main.

**Steps to reproduce**

The `service.name` is configured like the following https://github.com/open-telemetry/opentelemetry-cpp-contrib/blob/946b3d8e9576cdb5a078c88b1a007c574f23d42c/instrumentation/nginx/src/otel_ngx_module.cpp

**What is the expected behavior?**

The service name to be displayed in jaeger.

**What is the actual behavior?**

`OTLPResourceNoServiceName`

**Additional context**

**Comment by lalitb** [comment]
@johanneswuerbach - Can you confirm the opentelemetry-cpp release / commit used to test this. 
fyi - required support was added here : https://github.com/open-telemetry/opentelemetry-cpp/pull/758/files

**Comment by johanneswuerbach** [comment]
This was tested using `v0.6.0`, which doesn't include the linked PR. Is another release planned in the near future or better build from that specific commit?

**Comment by lalitb** [comment]
Yes, there is a new release coming up today with the fix : https://github.com/open-telemetry/opentelemetry-cpp/pull/800.

**Comment by lalitb** [comment]
Release is available here : https://github.com/open-telemetry/opentelemetry-cpp/releases/tag/v0.7.0

**Comment by johanneswuerbach** [comment]
Bumping to v0.7.0 solved this 😍
